### DISCUSSION TITLE
Missing functionality: `useWatchContractEvent` missing in Vue?

### DISCUSSION THREAD

**Discussion by ThatOneCalculator** [wevm/wagmi#4246] [Ideas]
### Check existing issues

- [X] I checked there isn't [already an issue](https://github.com/wevm/wagmi/issues) for the bug I encountered.

### Describe the bug

It seems that `useWatchContractEvent` exists in the React library but not the Vue one, and watching for contract events is important functionality.

### What Wagmi package(s) are you using?

@wagmi/vue

### Wagmi Package(s) Version(s)

0.0.40

### Viem Version

2.13.4

### TypeScript Version

5.5.4

**Comment by jxom** [comment]
Needs to be implemented. Moving to discussion as a feature request.

**Reply by ThatOneCalculator** [reply]
Fuck it, we ball. Working on it here: https://github.com/ThatOneCalculator/wagmi/tree/feat/vue-contract-event

**Comment by tmm** [comment]
https://wagmi.sh/vue/api/composables/useWatchContractEvent
